ui/src/app/namespaces/namespacesApi.ts

```diff
@@ -75,37 +75,48 @@ export const selectCurrentNamespace = createSelector(
 export const namespaceApi = createApi({
   reducerPath: 'namespaces-api',
   baseQuery,
-  tagTypes: ['Namespace'],
+  tagTypes: ['Namespace', 'Flag', 'Segment'],
   endpoints: (builder) => ({
     // get list of namespaces
-    listNamespaces: builder.query<INamespaceList, void>({
-      query: () => '/namespaces',
+    listNamespaces: builder.query<INamespaceList, { environmentKey: string }>({
+      query: ({ environmentKey }) => `/${environmentKey}/namespaces`,
```

Comment:
i realize im using `environmentKey` everywhere when the actual field on `environment` is `name`. 

@GeorgeMac are we set on `name` for environment? is there any checking that its a single string, lowercase, etc?

Reply:
We can definitely rename it to `key` if we want. We can add validation for that too.